sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/BasePartitionProcessor.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Messaging.EventHubs.Processor
+{
+    /// <summary>
+    ///   Processes events received from the Azure Event Hubs service.  An instance of this class or of a derived class
+    ///   will be created by the associated <see cref="EventProcessor{T}" /> for every partition it owns.  This class does
+    ///   not perform any kind of processing by itself and a useful partition processor is expected to be derived from it.
+    /// </summary>
+    ///
+    public class BasePartitionProcessor
+    {
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="BasePartitionProcessor"/> class.
+        /// </summary>
+        ///
+        public BasePartitionProcessor()
+        {
+        }
+
+        /// <summary>
+        ///   Initializes the partition processor.
+        /// </summary>
+        ///
+        /// <param name="partitionContext">Contains information about the partition this partition processor will be processing events from.  It's also responsible for the creation of checkpoints.</param>
+        /// 
+        /// <returns>A task to be resolved on when the operation has completed.</returns>
+        ///
+        public virtual Task InitializeAsync(PartitionContext partitionContext)
```

Comment:
I believe one of the points in replacing the interface with the abstract class was to allow users to only implement the methods they needed, as things starts to get overwhelming at first. The methods were supposed to do some kind of logging, but we currently don't have that in Event Processor, so there's nothing to do.

Reply:
Fair enough.  The class should probably still be `abstract` and I'd advocate for the process and error methods as well.   I can see the case for Initialize and Stop to be no-ops.

Reply:
I'm still a bit against marking it as `abstract`, at least for this preview. In case the user decides to use the factory approach to create partition processors, they would still need to create a useless type only to instantiate the generic `EventProcessor`, as its underlying type can't be abstract.

Reply:
I'm good with Initialize and Stop being just `virtual`, but I think that we need to force `ProcessEvents` and `ProcessError` to be implemented.